1、事务交给spring管理后，异常一定不能捕获。不然出现异常，事务都不回滚。如下代码：


@Service
public class UserDaoService {
	@Resource UserDao userDao;
	
	@Transactional(rollbackFor = Exception.class)
	public void save(String ss) throws Exception{
		String s = null;
		userDao.save(ss);
		userDao.save("abc");
		//int a = 1/0;
		throw new Exception("abre");//此处一定不能捕获
	}
	
	public List<User> getUsers(){
		return userDao.getUsers();
	}
}

2、事务交给spring管理后，session一定要通过factory.getCurrentSession()获得。不能通过factory.openSession()获得。通过前者获得事务才起作用，
   后者事务无效。
